source/server/active_tcp_socket.h

```diff
@@ -0,0 +1,227 @@
+public:
+  ActiveStreamListenerBase(Network::ConnectionHandler& parent, Event::Dispatcher& dispatcher,
+                           Network::ListenerPtr&& listener, Network::ListenerConfig& config);
+  static void emitLogs(Network::ListenerConfig& config, StreamInfo::StreamInfo& stream_info);
+
+  virtual void incNumConnections() PURE;
+  virtual void decNumConnections() PURE;
+
+  Event::Dispatcher& dispatcher() { return dispatcher_; }
+
+  /**
+   * Create a new connection from a socket accepted by the listener.
+   */
+  virtual void newConnection(Network::ConnectionSocketPtr&& socket,
+                             std::unique_ptr<StreamInfo::StreamInfo> stream_info) PURE;
+
+  virtual Network::BalancedConnectionHandlerOptRef
+  getBalancedHandlerByAddress(const Network::Address::Instance& address) PURE;
+
+  void onSocketAccepted(std::unique_ptr<ActiveTcpSocket> active_socket) {
+    // Create and run the filters
+    config_->filterChainFactory().createListenerFilterChain(*active_socket);
+    active_socket->continueFilterChain(true);
+
+    // Move active_socket to the sockets_ list if filter iteration needs to continue later.
+    // Otherwise we let active_socket be destructed when it goes out of scope.
+    if (active_socket->iter_ != active_socket->accept_filters_.end()) {
+      active_socket->startTimer();
+      LinkedList::moveIntoListBack(std::move(active_socket), sockets_);
+    } else {
+      if (!active_socket->connected_) {
+        // If active_socket is about to be destructed, emit logs if a connection is not created.
+        if (active_socket->stream_info_ != nullptr) {
+          emitLogs(*config_, *active_socket->stream_info_);
+        } else {
+          // If the active_socket is not connected, this socket is not promoted to active
+          // connection. Thus the stream_info_ is owned by this active socket.
+          ENVOY_BUG(active_socket->stream_info_ != nullptr,
+                    "the unconnected active socket must have stream info.");
+        }
+      }
+    }
+  }
+
+  Network::ConnectionHandler& parent_;
+  Event::Dispatcher& dispatcher_;
+  Network::ListenerPtr listener_;
+  const std::chrono::milliseconds listener_filters_timeout_;
+  const bool continue_on_listener_filters_timeout_;
+  std::list<std::unique_ptr<ActiveTcpSocket>> sockets_;
+};
+
+template <typename ActiveConnectionType>
+class TypedActiveStreamListenerBase : public ActiveStreamListenerBase {
+
+public:
+  using ActiveConnectionCollectionType = typename ActiveConnectionType::CollectionType;
+  TypedActiveStreamListenerBase(Network::ConnectionHandler& parent, Event::Dispatcher& dispatcher,
+                                Network::ListenerPtr&& listener, Network::ListenerConfig& config)
+      : ActiveStreamListenerBase(parent, dispatcher, std::move(listener), config) {}
+  using ActiveConnectionPtr = std::unique_ptr<ActiveConnectionType>;
+  using ActiveConnectionsPtr = std::unique_ptr<ActiveConnectionCollectionType>;
```

Comment:
Change ActiveConnectionsPtr to ActiveConnectionCollectionPtr

Two names that only differ from an 's' somewhere in the middle are too hard to tell apart.  Let's make the easier for humans to read.

Reply:
Good point!